blueprints/addon/index.js

```diff
@@ -38,7 +38,7 @@ module.exports = {
     contents.devDependencies['ember-disable-prototype-extensions'] = '^1.1.0';
 
     // add `ember-try` to addons by default
-    contents.devDependencies['ember-try'] = '~0.0.8';
+    contents.devDependencies['ember-try'] = '^0.1.2';
```

Comment:
See https://docs.npmjs.com/misc/semver#caret-ranges-1-2-3-0-2-5-0-0-4 for details on caret ranges.